resources/netlog_defs.ts

```diff
@@ -29,9 +31,9 @@ export type LogDefinition = {
   // a list of fields that are ok to not appear (or have invalid ids)
   optionalFields?: readonly number[];
 };
-export type LogDefinitionMap = { [name: string]: LogDefinition };
+export type LogDefinitionMap = { [lang in Lang]: { [name: string]: LogDefinition } };
 
-const logDefinitions = {
+const intlLogDefinitions = {
```

Comment:
The reason I didn't opt to go for by-version to begin with is that it would necessitate some sort of casting and version translation, e.g. detect parser version `2.6.4.1` on lang 'en' => use defs version `2.6.2.0`.

This would necessitate having a strict type, e.g:
```ts
const logDefinitions = {
  '2.2.1.0': {
    GameLog: {/*...*/},
  },
  '2.6.2.0': {
    GameLog: {/*...*/},
  },
} as const;
type ParserVersion = keyof typeof logDefinitions;
// Take a parser version number, such as 2.6.4.1, and a lang, such as 'en', and return the correct version, e.g. '2.6.2.0'
const toVersion = (version: string, lang: Lang): ParserVersion => {};
```

It's not feasible to map every single parser version to a key on the defs object, that'll cause issues with the template expansion of the net_fields.d.ts file, which expands out to everything else, due to the template possibilities exceeding built-in typescript limits (which exist for compiler performance reasons, IIRC).

Reply:
> It's not feasible to map every single parser version to a key on the defs object

Yeah, definitely agreed.  My expectation is that we'd only have keys for parser versions where the logs changed (so right now we'd really only have two keys).

> The reason I didn't opt to go for by-version to begin with is that it would necessitate some sort of casting and version translation, e.g. detect parser version 2.6.4.1 on lang 'en' => use defs version 2.6.2.0.

I think in the future this would be true if we wanted to make the emulator and tools handle this, or if we wanted to somehow do this based on the actual plugin version vs the parsed language version.  In the short term, I could imagine a simple parser language -> version lookup and it could get smarter from there.  (This is a minor point, but I also don't think language is an input to this? My understanding is that the Korean and Chinese users are using an older parser version at the moment, and so parser version is sufficient to determine the defs.)

> `const toVersion = (version: string, lang: Lang): ParserVersion => {};`

Yeah, I think this is exactly what I was expecting.  This doesn't bother me too much, because I don't think it proliferates in cactbot code too much, unless I am misunderstanding.  I imagine that the templated code would change language->version in this PR and then a follow up PR would call this `toVersion` function (with a simple implementation, maybe only based on lang) and set static variables appropriately.

Could you expand on your concerns here, re: "solution isn't as clean/elegant for the majority of the cactbot code"?